wallet/src/ui/app/redux/slices/app/index.ts

```diff
@@ -1,24 +1,35 @@
 // Copyright (c) 2022, Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
 
 import { AppType } from './AppType';
 import { DEFAULT_API_ENV } from '_app/ApiProvider';
 
 import type { PayloadAction } from '@reduxjs/toolkit';
 import type { API_ENV } from '_app/ApiProvider';
+import type { AppThunkConfig } from '_store/thunk-extras';
 
 type AppState = {
     appType: AppType;
     apiEnv: API_ENV | null;
+    showHideNetwork: boolean;
 };
 
 const initialState: AppState = {
     appType: AppType.unknown,
     apiEnv: DEFAULT_API_ENV,
+    showHideNetwork: false,
 };
 
+export const changeRPCNetwork = createAsyncThunk<void, void, AppThunkConfig>(
```

Comment:
I think I added a comment for this, using a middleware to change the api provider when we dispatch a `setApiEnv` action will make it easier to handle also the problems that will create. Clear current state for the 
previous network etc.